src/Containers/StoreHoursList/StoreHoursList.tsx

```diff
@@ -0,0 +1,550 @@
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import { BusinessTime } from '@styled-icons/fa-solid/BusinessTime';
+import { Edit } from '@styled-icons/boxicons-regular/Edit';
+import { Add } from '@styled-icons/ionicons-outline/Add';
+import { CircleWithCross } from '@styled-icons/entypo/CircleWithCross';
+import { Alert } from './Alert';
+import { SettingsCard } from '../SettingsCard';
+import { Modal } from '../Modal';
+import { Tag } from '../Tag';
+import { Heading, SmallText } from '../../Text';
+// eslint-disable-next-line import/no-cycle
+import { Button, Checkbox, Select, Timepicker, Input } from '../../Inputs';
+import { Mixins } from '../../Utils';
+import { MainInterface, ResponsiveInterface } from '../../Utils/BaseStyles';
+
+interface ICategoryWithHoursTypes {
+    category: string,
+    hoursByDay: {
+        monday: { to: string, from: string }[],
+        tuesday: { to: string, from: string }[],
+        wednesday: { to: string, from: string }[],
+        thursday: { to: string, from: string }[],
+        friday: { to: string, from: string }[],
+        saturday: { to: string, from: string }[],
+        sunday: { to: string, from: string }[]
+    }
+};
+
+interface I_DICT {
+    [key: string]: {
+        [key: string]: string; 
+    };
+}
+
+interface ITimeTypes {
+    to: Date | string 
+    from: Date | string 
+};
+
+interface StoreHoursListProps 
+    extends MainInterface, 
+        ResponsiveInterface, 
+        React.HTMLAttributes<HTMLDivElement> {
+            allCategories: ICategoryWithHoursTypes[],
+            constants: I_DICT
+};
+
+export const StoreHoursList: React.FC<StoreHoursListProps> = ({
+    allCategories,
+    constants
+}): React.ReactElement => {
+    const editModal = useState(false);
+    const [editModalState, setEditModalState] = editModal
```

Comment:
I get the following error when I do this:
![image](https://user-images.githubusercontent.com/62865150/93006806-df303280-f52e-11ea-94d3-2f5b8ca54beb.png)
I think it might be because the state is on when you click a button and thus that makes the state 'iterable'.

Reply:
> Why? Just do it on the same line to reduce code

modal takes state as itself, so u need to deconstruct/index the array to use the state/dispatch

Reply:
Could you provide a sample code of how to do this? I've never seen useState being restructured before.

Reply:
> Could you provide a sample code of how to do this? I've never seen useState being restructured before.

no wat ur doing is fine, theres nothing wrong with it. We cant reconstruct it or else React would detect a new array every re-render, which would fk the performance of the app 100%